common/djangoapps/third_party_auth/models.py

```diff
@@ -718,6 +731,25 @@ class Meta:
         verbose_name = "Provider Configuration (SAML IdP)"
         verbose_name_plural = "Provider Configuration (SAML IdPs)"
 
+    # validator method on row create
+    # looks at entity ID and slug
+    # check if any other records exist with that entity ID under a different slug
+    # prevents creation of new entry if other records found
+
+    def save(self, *args, **kwargs):
+        # Disallowing any new entries that have the same entity ID as an existing provider config
+        # Unless the slug matches.
+        # This both allows for the old architecture to create new rows on save but also prevents enterprise users from
+        # creating new configs that share entity ID's with other enterprises
+        # One consequence of this is that once a provider configuration is created, the slug is essentially locked in
```

Comment:
entity ID is changeable, the comment was referring to slug. I believe that you cannot change your slug once the config is created and saved in the DB (unless the entity ID is also changed simultaneously) because the model will generate a new row with that original entity ID and a new slug which would trigger the integrity error

Reply:
and yes, I should add a comment about entity ID's being "locked" by a customer according to their slug

Reply:
The whole slug/entity_id thing has been a factor in many rounds of troubleshooting over the past several years.